# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/svt-jpeg-xs/system-cpuinfo.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- SVT-JPEG-XS-0.9.0/Source/App/SampleDecoder/CMakeLists.txt.vanilla	2026-10-02 19:44:49.540048869 +0300
+++ SVT-JPEG-XS-0.9.0/Source/App/SampleDecoder/CMakeLists.txt	2026-10-02 19:47:49.074778002 +0300
@@ -19,6 +19,11 @@
 add_executable(SvtJpegxsSampleDecoder
     ${all_files})
 
+find_package(cpuinfo)
+
+target_link_libraries(SvtJpegxsSampleDecoder
+	    cpuinfo::cpuinfo)
+
 #Link only with required libraries to detect in linker time references between decoder_lib an encoder_lib
 set(lib_list_decoder
     $<TARGET_OBJECTS:COMMON_CODEC>
@@ -28,7 +33,6 @@
     $<TARGET_OBJECTS:DECODER_ASM_AVX512>
     $<TARGET_OBJECTS:DECODER_ASM_AVX2>
     $<TARGET_OBJECTS:DECODER_ASM_SSE4_1>
-    cpuinfo_public
     )
 
 #********** SET COMPILE FLAGS************
@@ -49,7 +53,6 @@
     DECODER_CODEC
     DECODER_ASM_AVX512
     DECODER_ASM_AVX2
-    DECODER_ASM_SSE4_1
-    cpuinfo_public)
+    DECODER_ASM_SSE4_1)
 
 install(TARGETS SvtJpegxsSampleDecoder RUNTIME DESTINATION ${CMAKE_INSTALL_FULL_BINDIR})
--- SVT-JPEG-XS-0.9.0/Source/App/SampleEncoder/CMakeLists.txt.vanilla	2026-10-02 19:42:31.904590056 +0300
+++ SVT-JPEG-XS-0.9.0/Source/App/SampleEncoder/CMakeLists.txt	2026-10-02 19:48:30.019286017 +0300
@@ -19,6 +19,11 @@
 add_executable(SvtJpegxsSampleEncoder
     ${all_files})
 
+find_package(cpuinfo)
+
+target_link_libraries(SvtJpegxsSampleEncoder
+	cpuinfo::cpuinfo)
+
 #Link only with required libraries to detect in linker time references between decoder_lib an encoder_lib
 set(lib_list_encoder
     $<TARGET_OBJECTS:COMMON_CODEC>
@@ -29,7 +34,6 @@
     $<TARGET_OBJECTS:ENCODER_ASM_SSE4_1>
     $<TARGET_OBJECTS:ENCODER_ASM_AVX2>
     $<TARGET_OBJECTS:ENCODER_ASM_AVX512>
-    cpuinfo_public
     )
 
 #********** SET COMPILE FLAGS************
@@ -51,7 +55,6 @@
     ENCODER_ASM_SSE2
     ENCODER_ASM_SSE4_1
     ENCODER_ASM_AVX2
-    ENCODER_ASM_AVX512
-    cpuinfo_public)
+    ENCODER_ASM_AVX512)
 
 install(TARGETS SvtJpegxsSampleEncoder RUNTIME DESTINATION ${CMAKE_INSTALL_FULL_BINDIR})
--- SVT-JPEG-XS-0.9.0/Source/Lib/CMakeLists.txt.vanilla	2026-10-02 19:49:36.797834775 +0300
+++ SVT-JPEG-XS-0.9.0/Source/Lib/CMakeLists.txt	2026-10-02 19:50:26.221015953 +0300
@@ -73,12 +73,14 @@
     $<TARGET_OBJECTS:DECODER_ASM_SSE4_1>
     )
 
+find_package(cpuinfo)
+
 set_target_properties(SvtJpegxsLib PROPERTIES OUTPUT_NAME "SvtJpegxs")
 set_target_properties(SvtJpegxsLib PROPERTIES VERSION ${SVT_LIB_VERSION})
 set_target_properties(SvtJpegxsLib PROPERTIES SOVERSION ${SVT_LIB_VERSION_MAJOR})
 set_target_properties(SvtJpegxsLib PROPERTIES C_VISIBILITY_PRESET hidden)
 target_link_libraries(SvtJpegxsLib PUBLIC ${PLATFORM_LIBS})
-target_link_libraries(SvtJpegxsLib PRIVATE cpuinfo_public)
+target_link_libraries(SvtJpegxsLib PRIVATE cpuinfo::cpuinfo)
 
 set(LIB_PKG_CONFIG_EXTRA_CFLAGS "")
 if (BUILD_SHARED_LIBS)
--- SVT-JPEG-XS-0.9.0/CMakeLists.txt.vanilla	2026-10-02 19:40:18.063370450 +0300
+++ SVT-JPEG-XS-0.9.0/CMakeLists.txt	2026-10-02 19:45:37.175602158 +0300
@@ -338,6 +338,7 @@
     #enable_testing()
 endif()
 
-add_subdirectory(third_party/cpuinfo)
+find_package(cpuinfo)
+#add_subdirectory(third_party/cpuinfo)
 
 install(DIRECTORY ${PROJECT_SOURCE_DIR}/Source/API/ DESTINATION "${CMAKE_INSTALL_FULL_INCLUDEDIR}/svt-jpegxs" FILES_MATCHING PATTERN "*.h")
